src/campaigns/notificationsPromoCampaign.ts

```diff
@@ -0,0 +1,63 @@
+import { MMKV } from 'react-native-mmkv';
+import {
+  Campaign,
+  CampaignCheckType,
+  CampaignKey,
+  GenericCampaignCheckResponse,
+} from './campaignChecks';
+import { RainbowWallet } from '@/model/wallet';
+import { Navigation } from '@/navigation';
+import { logger } from '@/logger';
+import store from '@/redux/store';
+import Routes from '@/navigation/routesNames';
+import { STORAGE_IDS } from '@/model/mmkv';
+
+const mmkv = new MMKV();
+
+export const notificationsCampaignAction = async () => {
```

Comment:
can this be generalized for all campaign types? i see that `swapsPromoCampaignAction` is identical except for the fact that its a different campaign

Reply:
Probably! Haven't thought too deeply on this logic. I think this sounds like a good follow-up PR though! Or maybe one we add a third promo, we'll know if they're always going to be similar enough to warrant consolidation.